internal/backends/python/python.go

```diff
@@ -248,10 +248,18 @@ func pythonMakeBackend(name string, python string) api.LanguageBackend {
 
 			return info
 		},
-		Add: func(pkgs map[api.PkgName]api.PkgSpec) {
+		Init: func(name string) {
 			if !util.Exists("pyproject.toml") {
-				util.RunCmd([]string{python, "-m", "poetry", "init", "--no-interaction"})
+				cmd := []string{python, "-m", "poetry", "init", "--no-interaction"}
+
+				if name != "" {
+					cmd = append(cmd, "--name", name)
+				}
+
+				util.RunCmd(cmd)
 			}
+		},
+		Add: func(pkgs map[api.PkgName]api.PkgSpec) {
```

Comment:
not sure but perhaps name should just be an arg to `Add` if it's not really used elsewhere.

Reply:
I hear your reasoning, @Zabot. We've tended to solve the problem right in front of us with notes on what to do in the future. If `upm add` is the only spot where `upm` will create a spec file, then let's pass the name as an argument to `Add` and leave a nice comment saying why we did that.